controllers/snapshot/snapshot_adapter.go

```diff
@@ -220,6 +221,9 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 
 		_, err = a.ensureEphemeralEnvironmentForScenarioExists(&integrationTestScenario, testStatuses, components, allEnvironments)
 		if err != nil {
+			if strings.Contains(err.Error(), strconv.Quote(integrationTestScenario.Spec.Environment.Name)+" not found") {
```

Comment:
Can `errors.IsNotFound(err)` be used here instead?

Reply:
No, because we also add some other info in the returned error [here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L784)

Reply:
Or we update [here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L784) to return `err` without additional info directly?

Reply:
But I hope to also add failure reason [here](https://github.com/redhat-appstudio/integration-service/pull/381/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R441) then report to GH ([here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L472-L474)) then user can know it.

Reply:
This got me thinking, we already get the list of existing environments with `a.loader.GetAllEnvironments()` [here](https://github.com/redhat-appstudio/integration-service/pull/381/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R205), we could use that to immediately determine if the environment exists for the scenario and skip calling ` a.ensureEphemeralEnvironmentForScenarioExists()` entirely.

Reply:
Yes, that is better than the current logic. Let me update it. Thanks!

Reply:
Updated. Thanks.